feature/bgp/gracefulrestart/otg_tests/bgp_graceful_restart_test/bgp_graceful_restart_test.go

```diff
@@ -0,0 +1,689 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bgp_graceful_restart_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/open-traffic-generator/snappi/gosnappi"
+	"github.com/openconfig/featureprofiles/internal/attrs"
+	"github.com/openconfig/featureprofiles/internal/deviations"
+	"github.com/openconfig/featureprofiles/internal/fptest"
+	"github.com/openconfig/featureprofiles/internal/otgutils"
+	gnps "github.com/openconfig/gnoi/system"
+
+	"github.com/openconfig/ondatra"
+	"github.com/openconfig/ondatra/gnmi"
+	"github.com/openconfig/ondatra/gnmi/oc"
+	"github.com/openconfig/ygnmi/ygnmi"
+	"github.com/openconfig/ygot/ygot"
+)
+
+func TestMain(m *testing.M) {
+	fptest.RunTests(m)
+}
+
+// The testbed consists of ate:port1 -> dut:port1 and
+// dut:port2 -> ate:port2.  The first pair is called the "source"
+// pair, and the second the "destination" pair.
+//
+//   * Source: ate:port1 -> dut:port1 subnet 192.0.2.0/30 2001:db8::192:0:2:0/126
+//   * Destination: dut:port2 -> ate:port2 subnet 192.0.2.4/30 2001:db8::192:0:2:4/126
+//
+// Note that the first (.0, .3) and last (.4, .7) IPv4 addresses are
+// reserved from the subnet for broadcast, so a /30 leaves exactly 2
+// usable addresses. This does not apply to IPv6 which allows /127
+// for point to point links, but we use /126 so the numbering is
+// consistent with IPv4.
+//
+
+const (
+	trafficDuration          = 1 * time.Minute
+	grTimer                  = 2 * time.Minute
+	triggerGrTimer           = 60
+	stopDuration             = 45 * time.Second
+	grRestartTime            = 120
+	grStaleRouteTime         = 120
+	ipv4SrcTraffic           = "192.0.2.2"
+	ipv6SrcTraffic           = "2001:db8::192:0:2:2"
+	ipv4DstTrafficStart      = "203.0.113.1"
+	ipv4DstTrafficEnd        = "203.0.113.2"
+	ipv6DstTrafficStart      = "2001:db8::203:0:113:1"
+	ipv6DstTrafficEnd        = "2001:db8::203:0:113:2"
+	advertisedRoutesv4Net    = "203.0.113.1"
+	advertisedRoutesv6Net    = "2001:db8::203:0:113:1"
+	advertisedRoutesv4Net2   = "203.0.113.3"
+	advertisedRoutesv6Net2   = "2001:db8::203:0:113:3"
+	advertisedRoutesv4Prefix = 32
+	advertisedRoutesv6Prefix = 128
+	aclNullPrefix            = "0.0.0.0/0"
+	aclv6NullPrefix          = "::/0"
+	aclName                  = "BGP-DENY-ACL"
+	aclv6Name                = "ipv6-policy-acl"
```

Comment:
Probably not required as no ACLs are configured?

Reply:
Indeed...removed